azure/compute/src/main/java/org/jclouds/azurecompute/AzureComputeApi.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c)Microsoft Open Technologies, Inc.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import org.jclouds.azurecompute.features.*;
+
+import javax.inject.Inject;
+
+public class AzureComputeApi implements Closeable {
+    private final VirtualMachineApi virtualMachineApi;
+    private final ImageApi imageApi;
+    private final LocationApi locationApi;
+    private final HostedServiceApi hostedServiceApi;
+    private final RoleSizeApi roleSizeApi;
+
+    @Inject
+    public AzureComputeApi(VirtualMachineApi virtualMachineApi, ImageApi imageApi, LocationApi locationApi,
```

Comment:
General: Following the Guice best practices, make all injection constructors package-private. This way only the injector (and the test classes) will be able to call it and you won't need to check for nulls, and other preconditions in the constructor code.